## Assignment #1 DATA 730

This assignment tests that you have the course tech stack set up. If you do, you will be able to complete this notebook and turn-in your work by pushing to your assignment repository.

## Tech stack checklist

*Fill out the checklist below once you have completed each task:*

- [✓] GitHub account
- [✓] SageMaker Studio Lab account

**Create the `data730` conda environment by right clicking the `environment.yml` file from the repo in SageMaker Studio Lab and selecting `Build conda Environment`.** 

In [4]:
# uncomment the following line and execute this cell to demonstrate you have created the data730 conda environment
system('conda env list | grep data730', intern = TRUE)

[1] "data730-lecture       *  /home/studio-lab-user/.conda/envs/data730-lecture"

In [5]:
#uncomment the following line and execute this cell to demonstrate that you have installed the tidymodels package
library('tidymodels')

## Let's do some statistical inference!

Read [this article](https://www.theguardian.com/world/2002/jan/04/euro.eu2). 

Execute the code below to investigate whether the coin is fair...

In [6]:
# These functions simulate tossing a fair coin n_tosses number of times for n_iterations number of iterations

toss_fair_coin_n_times = function(n_tosses = 250) {
    result = ifelse(runif(n = n_tosses) < 0.5, 'heads', 'tails')
    result
}

repeat_the_experiment = function(n_iterations, n_tosses = 250) {
    n_heads_each_iteration = c()
    for (i in 1:n_iterations) {
        n_heads = sum(toss_fair_coin_n_times(n_tosses) == 'heads')
        n_heads_each_iteration = c(n_heads_each_iteration, n_heads)
    } 
    n_heads_each_iteration
}

In [7]:
# Let's do 10,000 experiments of 250 tosses and see how often our outcome deviates from what was actually observed (139 heads)

n_iterations = 10000
n_tosses = 250
observed_deviation = 139 - 125 # 139 heads observed, expected value for a fair coin of 125

n_heads_each_experiment = repeat_the_experiment(n_iterations, n_tosses)

x = mean(abs(n_heads_each_experiment - 125) >= observed_deviation)
paste0('With a fair coin we would expect a deviation as large or larger than what we observed ', x*100, '% of times')

[1] "With a fair coin we would expect a deviation as large or larger than what we observed 8.84% of times"

**Do you think the coin was fair?** (Answer below in a markdown cell.)

Based off the article, I do not think the coin was fair. Two different entities discovered the coin landing heads at around 140 times compared to 125 what it should have been. These two entities were the Polish mathematicians Tomasz Gliszczynski and Waclaw Zawadowski and their students, finding the coin to land heads up 140 times, and the Guardian office who found the coin was heads up 139 times.

However, based off the code above, it shows that a fair coin would in fact deviate from the suspected "125 times heads up" that large or larger 8.84% of the time; this is a bigger percentage than I thought it was going to be. So almost 1 out of every 10 trials of 250 head tosses, a fair coin will have 139 heads or more. If the code proved that a fair coin would deviate "as large or larger than what we observed" 0.1% of times, then yes I would think that the one euro coin was definitely unfair. But I don't think the 8.84% is enough to say the one euro coin is unfair.